encoding/types.go

```diff
@@ -35,17 +36,19 @@ const (
 	CounterWithMetadatasType
 	BatchTimerWithMetadatasType
 	GaugeWithMetadatasType
+	TimedMetricWithForwardMetadataType
 )
 
 // UnaggregatedMessageUnion is a union of different types of unaggregated messages.
 // A message union may contain at most one type of message that is determined
 // by the `Type` field of the union, which in turn determines which one
 // of the field in the union contains the corresponding message data.
 type UnaggregatedMessageUnion struct {
-	Type                    UnaggregatedMessageType
-	CounterWithMetadatas    unaggregated.CounterWithMetadatas
-	BatchTimerWithMetadatas unaggregated.BatchTimerWithMetadatas
-	GaugeWithMetadatas      unaggregated.GaugeWithMetadatas
+	Type                           UnaggregatedMessageType
+	CounterWithMetadatas           unaggregated.CounterWithMetadatas
+	BatchTimerWithMetadatas        unaggregated.BatchTimerWithMetadatas
+	GaugeWithMetadatas             unaggregated.GaugeWithMetadatas
+	TimedMetricWithForwardMetadata aggregated.MetricWithForwardMetadata
```

Comment:
Is `aggregated` always `Timed`? If so we should consolidate the usage of "untimed"/"unaggregated" , "timed"/"aggregated"